### PR TITLE
gfx12 - change to use byte_sel modifier for v_cvt_f32_fp8 and v_cvt_f…

### PR DESCRIPTION
**Pull request by cmingch**
need to wait for compiler 6.4 promotion to mainline
for https://ontrack-internal.amd.com/browse/SWDEV-486447
compiler change op for 6.4
https://github.com/llvm/llvm-project/commit/57216f7bd64a78af6e037b34662ed49a6221734b#diff-7465cfa7f4559cb0f009e41ae43b585a8b3c17045446e01b13bb9698523afc18R402

### CODE REVIEW

File: tensilelite/Tensile/Components/GlobalWriteBatch.py [hunk 1]
```diff
@@ -2012,9 +2012,9 @@ def _addSumAlphaWithCBeta(self, kernel, ss, gwvw, elementIdx, vc0, tmpVgpr, cvtV
           # Generate single f32 code if edge is detected.
           isPK = False
           if ((vi + 1) == self.gwvw) and ((self.gwvw % 2) == 1):
-            if self.parentWriter.states.archCaps["NoSDWA"]: #cm review
+            if self.parentWriter.states.archCaps["NoSDWA"]:
```
Comment (jichangjichang) [inline tensilelite/Tensile/Components/GlobalWriteBatch.py]: Is it still suitable to use "NoSDWA"?

Comment (cmingch) [reply tensilelite/Tensile/Components/GlobalWriteBatch.py]: done

**Review by KKyang — DISMISSED** [review]

**Review by KKyang — DISMISSED** [review]

File: tensilelite/Tensile/Components/GlobalWriteBatch.py [hunk 2]
```diff
@@ -1993,9 +1993,15 @@ def _addSumAlphaWithCBeta(self, kernel, ss, gwvw, elementIdx, vc0, tmpVgpr, cvtV
...
           isPK = False
           if ((vi + 1) == self.gwvw) and ((self.gwvw % 2) == 1):
-            if self.parentWriter.states.archCaps["NoSDWA"]:
+            if self.parentWriter.states.archCaps["VOP3ByteSel"]:
               sb = 0 if self.gwvw == 1 else 1
-              module.add(VCvtFP8toF32(dst=vgpr(tmpVgpr), src=vgpr(dataV), vop3=VOP3PModifiers(op_sel=[0,sb])))
+              hipccver = globalParameters['HipClangVersion'].split(".")
+              hipccMaj = int(hipccver[0])
+              hipccMin = int(hipccver[1])
+              if not (hipccMaj >= 6 and hipccMin >= 4):
```
Comment (jichangjichang) [inline tensilelite/Tensile/Components/GlobalWriteBatch.py]: Are you sure that all hipccMin >= 4 version contains the compiler changes?
Do we have exact hipccPatch that compiler starts to support VOP3ByteSel?

Comment (cmingch) [reply tensilelite/Tensile/Components/GlobalWriteBatch.py]: as per compiler Mark's confirm, version 6.4 is enough.

File: tensilelite/Tensile/Components/GlobalWriteBatch.py [hunk 3]
```diff
@@ -2038,7 +2050,7 @@ def _addSumAlphaWithCBeta(self, kernel, ss, gwvw, elementIdx, vc0, tmpVgpr, cvtV
             if self.parentWriter.states.archCaps["NoSDWA"]:
               # Enable WORD_0 of 2-nd VGPR with vi=4 for vw=8
               sb = 0 if vi%4 == 0 else 1
-              module.add(VCvtPkFP8toF32(dst=vgpr(tmpVgpr, 2), src=vgpr(dataV), vop3=VOP3PModifiers(op_sel=[sb])))
+              module.add(VCvtPkBF8toF32(dst=vgpr(tmpVgpr, 2), src=vgpr(dataV), vop3=VOP3PModifiers(op_sel=[sb])))
```
Comment (jichangjichang) [inline tensilelite/Tensile/Components/GlobalWriteBatch.py:2051]: Don't need to modify this ?

Comment (cmingch) [reply tensilelite/Tensile/Components/GlobalWriteBatch.py:2051]: their change is not applied to v_cvt_pk_f32_fp8/bf8

Comment (cmingch) [reply tensilelite/Tensile/Components/GlobalWriteBatch.py:2051]: image

File: tensilelite/Tensile/Components/GlobalWriteBatch.py (hunk 2 above)
Comment (jichangjichang) [reply tensilelite/Tensile/Components/GlobalWriteBatch.py]: I only saw Mark suggest us checking Clang version.
Have you verified it in internal docker by only check hipcc >= 6.4?

File: tensilelite/Tensile/Components/GlobalWriteBatch.py (hunk 3 above)
Comment (jichangjichang) [reply tensilelite/Tensile/Components/GlobalWriteBatch.py:2051]: got it

File: tensilelite/Tensile/Components/GlobalWriteBatch.py (hunk 2 above)
Comment (jichangjichang) [inline tensilelite/Tensile/Components/GlobalWriteBatch.py]: should be 
not ((hipccMaj == 6 and hipccMin >= 4) or hipccMaj >= 7)

File: tensilelite/Tensile/Common.py [hunk 4]
```diff
@@ -1764,6 +1765,9 @@ def assignGlobalParameters( config ):
       if 'HIP version' in line:
         globalParameters['HipClangVersion'] = line.split()[2]
         print1("# Found  hipcc version " + globalParameters['HipClangVersion'])
+      if 'AMD clang version' in line:
+        globalParameters['AMDClangVersion'] = line.split()[3]
+        print1("# Found  clang version " + globalParameters['AMDClangVersion'])
```
Comment (jichangjichang) [inline tensilelite/Tensile/Common.py:1770]: it is better to use clang tool directly to check the version.
If user try to use upstream clang, hipcc won't be there.

**Review by jichangjichang — APPROVED** [review]

**Review by KKyang — APPROVED** [review]

**Pull request merged**